worldedit-bukkit/src/main/java/com/sk89q/worldedit/bukkit/adapter/Regenerator.java

```diff
@@ -0,0 +1,559 @@
+        SequentialTasks<ConcurrentTasks<SequentialTasks<Long>>> tasks;
+        if (maxz - minz > maxx - minx) {
+            int numlists = Math.min(requiredneighbors * 2 + 1, maxx - minx + 1);
+
+            Int2ObjectOpenHashMap<SequentialTasks<Long>> byx = new Int2ObjectOpenHashMap();
+            int expectedListLength = (allcoords.size() + 1) / (maxx - minx);
+
+            //init lists
+            for (int i = minx; i <= maxx; i++) {
+                byx.put(i, new SequentialTasks(expectedListLength));
+            }
+
+            //sort into lists by x coord
+            for (Long xz : allcoords) {
+                byx.get(MathMan.unpairIntX(xz)).add(xz);
+            }
+
+            //create parallel tasks
+            tasks = new SequentialTasks(numlists);
+            for (int offset = 0; offset < numlists; offset++) {
+                ConcurrentTasks<SequentialTasks<Long>> para = new ConcurrentTasks((maxz - minz + 1) / numlists + 1);
+                for (int i = 0; minx + i * numlists + offset <= maxx; i++)
+                    para.add(byx.get(minx + i * numlists + offset));
+                tasks.add(para);
+            }
+        } else {
+            int numlists = Math.min(requiredneighbors * 2 + 1, maxz - minz + 1);
+
+            Int2ObjectOpenHashMap<SequentialTasks<Long>> byz = new Int2ObjectOpenHashMap();
+            int expectedListLength = (allcoords.size() + 1) / (maxz - minz);
+
+            //init lists
+            for (int i = minz; i <= maxz; i++) {
+                byz.put(i, new SequentialTasks(expectedListLength));
+            }
+
+            //sort into lists by x coord
+            for (Long xz : allcoords) {
+                byz.get(MathMan.unpairIntY(xz)).add(xz);
+            }
+
+            //create parallel tasks
+            tasks = new SequentialTasks(numlists);
+            for (int offset = 0; offset < numlists; offset++) {
+                ConcurrentTasks<SequentialTasks<Long>> para = new ConcurrentTasks((maxx - minx + 1) / numlists + 1);
+                for (int i = 0; minz + i * numlists + offset <= maxz; i++)
+                    para.add(byz.get(minz + i * numlists + offset));
+                tasks.add(para);
+            }
+        }
+
+        return tasks;
+    }
+
+    protected static Random getChunkRandom(long worldseed, int x, int z) {
+        Random random = new Random();
+        random.setSeed(worldseed);
+        long xRand = random.nextLong() / 2L * 2L + 1L;
+        long zRand = random.nextLong() / 2L * 2L + 1L;
+        random.setSeed((long) x * xRand + (long) z * zRand ^ worldseed);
+        return random;
+    }
+
+    //util
+    public static void time(Runnable r, String text) {
+        long starttask = System.currentTimeMillis();
+//            System.out.println(text);
+        r.run();
+        System.out.println(text + " took " + (System.currentTimeMillis() - starttask) + "ms");
+    }
```

Comment:
ideally everything like this should be removed for review. This could also just have been done without runnables and in the actual method it was originally.

Reply:
I removed most of the sysouts and commented this out, since it might be useful if we ever need to debug it

Reply:
The extra overhead of the runnable isn't worth having one method to output the time the code took to run tbh, especially since it's won't actually be fully representative of the code snippet. Any runnables should be removed imo and just executed like normal code.

Reply:
done